lib/node-utils/index.ts

```diff
@@ -235,25 +234,26 @@ export function isPromiseHandled(nodeIdentifier: TSESTree.Identifier): boolean {
 }
 
 export function getVariableReferences(
-  context: RuleContext<string, []>,
+  context: TSESLint.RuleContext<string, []>,
   node: TSESTree.Node
 ): TSESLint.Scope.Reference[] {
   return (
     (ASTUtils.isVariableDeclarator(node) &&
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
       context.getDeclaredVariables(node)[0]?.references?.slice(1)) ||
```

Comment:
I don't know which condition is the rule complaining here.

Reply:
I think `slice` is always possible, as `references` will always be string/array?

Reply:
That seems to be the reason indeed! However, there are two optional chainings in the statement that could return a falsy value, and that's where the OR fallback would be necessary.

The rule doesn't seem to be aware of this? It seems by its type the rule assumes `context.getDeclaredVariables(node)` will return an array with at least one element, so `context.getDeclaredVariables(node)[0]` will always exist, but that's not the case.

I'm not sure now how "empty arrays" can be typed in TS, but I'd say that's the problem of this rule?

Reply:
It seems this is the reason indeed: https://github.com/typescript-eslint/typescript-eslint/issues/1798

I'm not if I want to enable this rule then. It could cause introducing errors by removing necessary conditions involving arrays.